Go term cosh (#3200)
Codecademy/docs: content/go/concepts/math-functions/terms/cosh/cosh.md

---
Title: 'Cosh()'
Description: 'Returns the hyperbolic cosine of a given value.'
Subjects:
  - 'Code Foundations'
  - 'Computer Science'
Tags:
  - 'Math'
  - 'Trigonometry'
  - 'Functions'
CatalogContent:
  - 'learn-go'
  - 'paths/computer-science'
---

The **`Cosh()`** of the `math` package is an inbuilt function which takes a given value and returns its hyperbolic consine value.

## Syntax

```pseudo
import "math"

CoshX = math.Cosh(x)
```

> Where `CoshX` is the hyperbolic cosine value of `x`, it is of type _float64_. It always return the hyperbolic consine value except some special cases:

Reviewer: ```suggestion
Where `CoshX` is the hyperbolic cosine value of `x`, it is of type _float64_. It always returns the hyperbolic cosine value except for some special cases:
```

> Where `CoshX` is the hyperbolic cosine value of `x`, it is of type _float64_. It always returns the hyperbolic cosine value except for some special cases:

Reviewer 2: ```suggestion
Where `CoshX` is the hyperbolic cosine value of `x`, it is of type `float64`. It always returns the hyperbolic cosine value except for some special cases:
```

> - CoshX of `Cosh(±0)` is `1`
> - CoshX of `Cosh(±Inf)` is `+Inf`
> - CoshX of `Cosh(NaN)` is `NaN`

Reviewer 2: ```suggestion
- `Cosh(±0)` is `1`
- `Cosh(±Inf)` is `+Inf`
- `Cosh(NaN)` is `NaN`
```

## Example

> The following Golang Program calculates the hyperbolic cosine of `x` and prints out the result:

Reviewer: ```suggestion
The following Golang program calculates the hyperbolic cosine of `x` and prints out the result:
```

```go
package main

import (
  "fmt"
  "math"
)

func main() {
  fmt.Println(math.Cosh(5))
  fmt.Println(math.Cosh(-5))
}
```

The output will be:

```shell
74.21
74.21
```

## Codebyte Example

> The following example shows how the `Cosh()` function handles float64 arguments including _special cases_ arguments.

Reviewer: ```suggestion
The following example shows how the `Cosh()` function handles float64 arguments including special cases.
```

> ```codebyte/golang

Reviewer 2: 2 of 3 of these results are duplicates from the previous example. Codebytes should not duplicate previous work. Ideally it should demonstrate some new functionality, information, etc. I would suggest updating the values or deleting the example.
